tests/integration_tests/scheduler/test_openpbs_driver.py

```diff
@@ -40,6 +50,71 @@ def test_openpbs_driver_with_poly_example(queue_name_config):
     )
 
 
+@pytest.mark.timeout(30)
+@pytest.mark.integration_test
+@pytest.mark.usefixtures("copy_poly_case")
+@pytest.mark.parametrize(
+    "text_to_ignore",
+    [
+        "pbs_iff: cannot connect to host\npbs_iff: all reserved ports in use",
+        "qstat: Invalid credential",
+    ],
+)
+def test_that_openpbs_driver_ignores_qstat_flakiness(
+    text_to_ignore, tmp_path, caplog, monkeypatch
+):
+    monkeypatch.chdir(tmp_path)
+    bin_path = tmp_path / "bin"
+    bin_path.mkdir()
+    monkeypatch.setenv("PATH", f"{bin_path}:{os.environ['PATH']}")
+    qsub_path = bin_path / "qsub"
+    qsub_path.write_text("#!/bin/sh\necho '1'")
+    qsub_path.chmod(qsub_path.stat().st_mode | stat.S_IEXEC)
+    qstat_path = bin_path / "qstat"
+    qstat_path.write_text(
+        "#!/bin/sh"
+        + dedent(
+            f"""
+            count=0
+            if [ -f counter_file ]; then
+                count=$(cat counter_file)
+            fi
+            echo "$((count+1))">counter_file
+            if [ $count -ge 3 ]; then
+                json_flag_set=false;
+                while [ "$#" -gt 0 ]; do
+                    case "$1" in
+                        -Fjson)
+                            json_flag_set=true
+                            ;;
+                    esac
+                    shift
+                done
+                if [ "$json_flag_set" = true ]; then
+                    echo '{json.dumps({"Jobs": {"1": {"Job_Name": "1", "job_state": "E", "Exit_status": "0"}}})}'
+                else
+                    echo "{QSTAT_HEADER}"; printf "{QSTAT_WIDE_FORMATTER}" 1 foo someuser 0 E normal
+                fi
+            else
+                echo "{text_to_ignore}">&2
+                exit 2
+            fi
+        """
+        )
+    )
+    qstat_path.chmod(qstat_path.stat().st_mode | stat.S_IEXEC)
+    with open("poly.ert", mode="a+", encoding="utf-8") as f:
+        f.write("QUEUE_SYSTEM TORQUE\nNUM_REALIZATIONS 1")
+    run_cli(
+        ENSEMBLE_EXPERIMENT_MODE,
+        "--enable-scheduler",
+        "poly.ert",
+    )
+    assert Path("counter_file").exists(), str(tmp_path)
+    assert int(Path("counter_file").read_text(encoding="utf-8")) >= 3
+    assert text_to_ignore not in caplog.text
```

Comment:
Here you assert that the text is not present in logged messages, but are you capturing stdout and/or stderr  (it could be that the code transfers stdout/stderr to logs, not sure) (there is also `capsys` I think).  

Are you able to poke the driver code ("remove the feature that mutes these message") and confirm that the test will fail?

Reply:
Yes. This retry functionality in polling does not output the stderr from `process.communicate(...)` anywhere.
```
stdout, stderr = await process.communicate()
if process.returncode not in {0, QSTAT_UNKNOWN_JOB_ID}:
       # Any unknown job ids will yield QSTAT_UNKNOWN_JOB_ID, but
       # results for other job ids on stdout can be assumed valid.
       await asyncio.sleep(_POLL_PERIOD)
       continue
```

Reply:
In the test you cannot rely on what the current code is doing, you are protecting the current code to be changed for the worse in the future. Thus, in the test you should assert that the string in the test cannot be found either captured logs or in captured stdout/stderr (or anywhere else if you can come up with something to test)